Tests/Functional/Command/CreateIndexCommandTest.php

```diff
@@ -64,6 +64,37 @@ public function testExecute($managerName, $arguments, $options)
     }
 
     /**
+     * Tests creating index in case of existing this index. Configuration from tests yaml.
+     *
+     * @param string $managerName
+     * @param array  $arguments
+     * @param array  $options
+     *
+     * @dataProvider getTestExecuteData
+     */
+    public function testExecuteWithIndexExistence($managerName, $arguments, $options)
```

Comment:
You should not use different test case data provider and try to behave inside test case to match expectations. To test `--if-not-exists` please write separate test case, the data provider is not necessary.